From ce9de08622dea7986c539d14a50bdad2c9379d75 Mon Sep 17 00:00:00 2001
From: You-Sheng Yang <vicamo.yang@canonical.com>
Date: Fri, 7 Jul 2023 18:38:23 +0800
Subject: [PATCH] backport: fix build against v6.5

Upstream commit 54d020692b34 ("mm/gup: remove unused vmas parameter from
get_user_pages()") targeting v6.5 removed the last argument to
get_user_pages.

Closes: #157
Signed-off-by: You-Sheng Yang <vicamo.yang@canonical.com>
---
 drivers/media/pci/intel/ipu-psys.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu-psys.c b/drivers/media/pci/intel/ipu-psys.c
index ca2830e5dc7e..6ac4cab1cb5c 100644
--- a/drivers/media/pci/intel/ipu-psys.c
+++ b/drivers/media/pci/intel/ipu-psys.c
@@ -218,7 +218,11 @@ static int ipu_psys_get_userpages(struct ipu_dma_buf_attach *attach)
 #else
 				    FOLL_WRITE,
 #endif
-				    pages, NULL);
+				    pages
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 5, 0)
+				    , NULL
+#endif
+				    );
 		if (nr < npages)
 			goto error_up_read;
 	}
